src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java

```diff
@@ -129,21 +138,43 @@ public PasswordResetLinkServiceImpl(ConfigurationClient configurationClient, Aut
         String generatedLink = linkHost + linkPath + '/' + token;
         linkHolder.value = generatedLink;
 
+        String expirationTimeFromConfig =
+          configMapHolder.value.getOrDefault(LINK_EXPIRATION_TIME_CONFIG_KEY, LINK_EXPIRATION_TIME_DEFAULT);
+        String expirationUnitOfTimeFromConfig = configMapHolder.value.getOrDefault(
+          LINK_EXPIRATION_UNIT_OF_TIME_CONFIG_KEY, LINK_EXPIRATION_UNIT_OF_TIME_DEFAULT);
+
         String eventConfigName = passwordExistsHolder.value ? RESET_PASSWORD_EVENT_CONFIG_NAME : CREATE_PASSWORD_EVENT_CONFIG_NAME;
         Notification notification = new Notification()
           .withEventConfigName(eventConfigName)
           .withRecipientId(userId)
           .withContext(
             new Context()
               .withAdditionalProperty("user", JsonObject.mapFrom(userHolder.value))
-              .withAdditionalProperty("link", generatedLink))
+              .withAdditionalProperty("link", generatedLink)
+              .withAdditionalProperty("expirationTime", expirationTimeFromConfig)
+              .withAdditionalProperty("expirationUnitOfTime", expirationUnitOfTimeFromConfig))
           .withText(StringUtils.EMPTY)
           .withLang(DEFAULT_NOTIFICATION_LANG);
         return notificationClient.sendNotification(notification, connectionParams);
       })
       .map(v -> linkHolder.value);
   }
 
+  private long convertDateToMilliseconds(String expirationTimeDefault, String expirationUnitOfTime) {
+    long expirationTime = Long.parseLong(expirationTimeDefault);
+    switch (expirationUnitOfTime) {
+      case "MINUTES":
+        return TimeUnit.MINUTES.toMillis(expirationTime);
+      case "HOURS":
+        return TimeUnit.HOURS.toMillis(expirationTime);
+      case "DAYS":
+        return TimeUnit.DAYS.toMillis(expirationTime);
+      case "WEEKS":
+        return TimeUnit.DAYS.toMillis(expirationTime) * 7;
+      default: throw new IllegalStateException("Can't convert time period to milliseconds");
```

Comment:
IllegalArgumentException might be a better choice.

Reply:
I added `UnprocessableEntityException`, ExceptionHelper handle this exception and return available response